File: 16-09-2022/Repeated_DNA_Sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        if len(s)<=10:
            return []
        d = {}
        l = []
        for i in range(10,len(s)+1):
            if s[i-10:i] not in d:
                d[s[i-10:i]]=1
            else:
                if d[s[i-10:i]]:
                    l.append(s[i-10:i])
                    d[s[i-10:i]] = 0 
        return l

File: Farhaan/Week-2 Problems/repeated-dna-sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
       vector<string> a;
        unordered_map<string,int> mp;
        if(s.size()<10)
            return a;       
        
        for(int i = 0;i<s.size()-9;i++){
            string r = s.substr(i,10);
            mp[r]++;
        }
        
        for(auto b :mp)
            if(b.second>1)
                a.push_back(b.first);
        return a;
    };
};